### DISCUSSION TITLE
Paste in lexical

### DISCUSSION THREAD

**Discussion by dunaevski** [facebook/lexical#2553] [Q&A] [answered]
Who can tell me, how I can get connected to the insertion, I need to create a custom plugin for inserting images via cmd + v and load to the server, then paste url in to img

**Comment by armelpingault** [comment]
I am also struggling to find something similar to what I am using with DraftJS to trigger an action depending on which content has been paste to the editor:
- https://draftjs.org/docs/api-reference-editor/#handlepastedtext
- https://draftjs.org/docs/api-reference-editor/#handlepastedfiles

**Comment by acywatson** [comment, marked as answer]
You can use the command system to do something in response to a paste action:

https://github.com/facebook/lexical/blob/5caf9231d456cadeb0f83fb69602b3bc6e433710/packages/lexical-rich-text/src/index.ts#L826

More on commands:

https://lexical.dev/docs/concepts/commands

**Reply by armelpingault** [reply]
Ah yeah thanks a lot, I missed it while reading the Concepts, something like this is finally working for me:
```
editor.registerCommand<ClipboardEvent>(
  PASTE_COMMAND,
  (event) => {
    const pastedText = event.clipboardData?.getData('Text');
        
    if (pastedText && pastedText === 'easter-egg') {
      editor.dispatchCommand(EXECUTE_EASTER_EGG, null);
      return true;
    }
        
    return false;
  },
  COMMAND_PRIORITY_EDITOR,
);
```
